### PR TITLE
A helper function to find the jvmci internal jdk8.

### PR DESCRIPTION
**Pull request by vext01**
OK?

### CODE REVIEW

**Comment by ltratt** [comment]
Can we give an idea of which directories? The comment is a bit vague...

**Comment by vext01** [comment]
Made the function name more accurate and documented it more. OK?

File: krun/vm_defs.py [hunk 1]
```diff
@@ -293,6 +294,30 @@ def check_benchmark_files(self, benchmark, entry_point):
...
+    or it could be:
+    jvmci/jdk1.8.0_66-internal/product/bin/java
+
+    This is a helper function to try and find the 'java' binary
+    inside this "moving" directory.
+
+    arguments:
+    base_dir -- base jvmci directory"""
+
+    for f in os.listdir(base_dir):
```
Comment (ltratt) [inline krun/vm_defs.py]: Could this be `for f in fnmatch.filter(os.listdir(base_dir), "jdk1.8*internal"):`?

Comment (ltratt) [inline krun/vm_defs.py]: Also `f` suggests "file"; maybe `l` for leaf or `p` for path?

**Comment by vext01** [comment]
We can use `fnmatch.filter`, but we wouldn't use it in a loop. In doing so we can also check there is only one match, thus making this a bit more robust.

Try this. OK to squish?

**Comment by ltratt** [comment]
I'm going to suggest we remove the pointless `assert`, squash, then merge. OK?

**Comment by vext01** [comment]
Hey,

On Tue, Sep 29, 2015 at 03:31:25AM -0700, Laurence Tratt wrote:
> I'm going to suggest we remove the pointless `assert`, squash, then merge. OK?

The conditional is not exhaustive, so I would argue the assert is good
practice. (OK, so we should never have a list of length <0, but it doesn't
hurt and krun doesn't need to be high-performance.)

Equally I don't care much either. We are both nitpicking at this point :)

-- 
Best Regards
Edd Barrett

http://www.theunixzoo.co.uk

**Comment by ltratt** [comment]
I am a fan of `assert`s in general, but I think we can go overboard. If the list has length less than 0, I'd be pretty worried ;) Let's kill it, squash, and then I can merge.

**Comment by vext01** [comment]
splat.

**Comment by vext01** [comment]
Don't forget the accompanying change here: https://github.com/softdevteam/warmup_experiment/pull/13

File: krun/vm_defs.py [hunk 2]
```diff
@@ -293,6 +294,34 @@ def check_benchmark_files(self, benchmark, entry_point):
...
+    This is a helper function to try and find the 'java' binary
+    inside this "moving" directory.
+
+    arguments:
+    base_dir -- base jvmci directory"""
+
+    matches = fnmatch.filter(os.listdir(base_dir), 'jdk1.8.0*internal*')
+
+    if len(matches) == 1:
+            return os.path.join(base_dir, matches[0], "product", "bin", "java")
```
Comment (ltratt) [inline krun/vm_defs.py]: Broken indentation?

**Comment by ltratt** [comment]
I'm not sure if I missed the indentation before, but we should fix it before merging.

**Comment by vext01** [comment]
go for it.

**Pull request merged**